File: Extensions/ShareExtension/ShareExtension/ViewController.m
//
//  ViewController.m
//  ShareExtension
//
//  Created by Hari Kunwar on 11/20/14.
//  Copyright (c) 2014 Hari Kunwar. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
{
    NSUserDefaults *_userDefault;
    __weak IBOutlet UILabel *_textLabel;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    _userDefault = [[NSUserDefaults alloc] initWithSuiteName:@"group.harikunwarDev"];
    NSString *text = [_userDefault objectForKey:@"textKey"];
    _textLabel.text = text;

    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(sharePost:)
                                                 name:NSUserDefaultsDidChangeNotification
                                               object:nil];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)sharePost:(NSNotification *)notification {
    NSString *text = [_userDefault objectForKey:@"textKey"];
    _textLabel.text = text;
}

@end

File: FacebookLogin/FacebookLogin/ViewController.swift
//
//  ViewController.swift
//  FacebookLogin
//
//  Created by Hari Kunwar on 2/23/15.
//  Copyright (c) 2015 learning. All rights reserved.
//

import UIKit

class ViewController: UIViewController, FBLoginViewDelegate {

    @IBOutlet weak var profilePictureView: FBProfilePictureView!
    @IBOutlet weak var loginView: FBLoginView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: FBLoginViewDelegate
    func loginViewFetchedUserInfo(loginView: FBLoginView!, user: FBGraphUser!) {
        profilePictureView.profileID = user.objectID
    }
    
    func loginViewShowingLoggedOutUser(loginView: FBLoginView!) {
        profilePictureView.profileID = nil
    }
}


